mlir/docs/RewritingMLIR.md

```diff
@@ -0,0 +1,129 @@
+# Rewriting MLIR
```

Comment:
How about calling this `Operation Builders and Rewriters`?

Reply:
Should I add a section about `OpBuilder`? That could get a bit involved, and I'm not as familiar with it since I didn't fight with it like I did with `RewriterBase`